### PR TITLE
Align implementation with base spec

### LINKED ISSUE #13: Pull in spec updates
These two changes need to be reflected in the implementation:

* https://github.com/ietf-rats/ietf-corim-cddl/pull/129
* https://github.com/ietf-rats/ietf-corim-cddl/pull/130

### PR DESCRIPTION
**Pull request by thomas-fossati**
* Add optional validity map to unsigned corim
    * https://github.com/ietf-rats/ietf-corim-cddl/pull/129

* Split signer role from manifest creator
    * https://github.com/ietf-rats/ietf-corim-cddl/pull/130

* Make Meta an opaque field in the COSE header
    * https://github.com/ietf-rats/ietf-corim-cddl/pull/133

Fixes #13

### CODE REVIEW

File: corim/meta.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
+// Copyright 2021 Contributors to the Veraison project.
```
Comment (yogeshbdeshpande) [inline corim/meta.go:1]: Perhaps we should clarify somewhere the use of word meta, is aligned with the CDDL schema and is used to carry information about the CoRIM Signer Entity?

File: corim/meta.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	return o
+}
+
+// Valid checks the validity of individual fields within Signer
+func (o Signer) Valid() error {
+	if o.Name == "" {
+		return errors.New("empty name")
+	}
+
+	if o.URI != nil {
```
Comment (yogeshbdeshpande) [inline corim/meta.go:57]: Should we check if it is a valid URI instead of just checking for a empty URI ?

File: corim/signedcorim.go [hunk 3]
```diff
@@ -49,13 +51,32 @@ func (o *SignedCorim) processHdrs() error {
...
+
 	return nil
 }
 
 // FromCOSE decodes and effects syntactic validation on the supplied
-// signed-corim message, including the embedded unsigned-corim.
+// signed-corim message, including the embedded unsigned-corim and corim-meta.
 // On success, the unsigned-corim-map is made available via the UnsignedCorim
-// field.
+// field while the corim-meta-map is decoded into the Meta field.
```
Comment (yogeshbdeshpande) [inline corim/signedcorim.go:79]: Nice description! Well done!

**Review by yogeshbdeshpande — COMMENTED** [review]
Have left few minor comments seeking clarification. In general looks ok to me!
Did some tidy up!

File: corim/meta.go (hunk 2 above)
Comment (thomas-fossati) [reply corim/meta.go:57]: good point!

File: corim/meta.go (hunk 1 above)
Comment (thomas-fossati) [reply corim/meta.go:1]: to that purpose I have expanded the description of the `Meta` type -- see 9277b88

File: corim/meta.go (hunk 2 above)
Comment (thomas-fossati) [reply corim/meta.go:57]: done in 122140e

File: corim/meta.go (hunk 1 above)
Comment (yogeshbdeshpande) [reply corim/meta.go:1]: yes, looks ok to me now!

**Comment by thomas-fossati** [comment]
> Did some tidy up!

Thanks for the inline edits!

**Review by yogeshbdeshpande — APPROVED** [review]

**Pull request merged**